include/swift/ClangImporter/ClangImporter.h

```diff
@@ -705,6 +705,9 @@ bool isCxxConstReferenceType(const clang::Type *type);
 /// Determine whether this typedef is a CF type.
 bool isCFTypeDecl(const clang::TypedefNameDecl *Decl);
 
+/// Determine whether type is a c++ foreign reference type.
+bool isForeignReferenceTypeWithoutImmortalAttrs(const clang::QualType type);
```

Comment:
I would slightly prefer a more descriptive name for this, e.g. `isRefCountedForeignReferenceType`, or `isSharedForeignReferenceType`. But I don't feel strongly about this.

Reply:
I can address this when I do some refactoring work where I plan to move helper/utility functions that query the classification of a clang type.

Reply:
Ref: rdar://138803759